### DISCUSSION TITLE
Retry a failed Workflow just from failed step/node

### DISCUSSION THREAD

**Discussion by bygui86** [argoproj/argo-workflows#12005] [Q&A] [answered]
Is it possible to retry a failed Workflow just from the step/node that failed?

For example I have a Workflow with following steps (in a dag):
1. build-code
2. test-code
3. build-container
4. deploy

For any reason, the build-container step fails... is it possible to retry the same Workflow starting directly from build-container step instead form the beginning?

**Comment by terrytangyuan** [comment, marked as answer]
That's exactly what the `argo retry` CLI command does. There are also flags like `--node-field-selector` and `--restart-successful` that you can use to select nodes to retry more specifically.

**Comment by bygui86** [comment]
Thanks @terrytangyuan do you have also a link to the documentation about those flags?
what about the UI? no option to have same result from UI?
CLI is nice, but I cannot ask everybody to install and use it as sometimes they are not really technical people...

**Reply by terrytangyuan** [reply]
UI may not have all the functionalities yet. If you need a particular one, feel free to submit an issue

**Reply by agilgur5** [reply]
For posterity, #12022 was opened for this functionality in the UI. There is actually an older issue that it duplicated though, #10139

**Comment by bygui86** [comment]
@terrytangyuan  I found the documentation [here](https://argoproj.github.io/argo-workflows/cli/argo_retry/) but it's pretty short and not really intuitive...
`--node-field-selector string   selector of nodes to reset, eg: --node-field-selector inputs.paramaters.myparam.value=abc`
so I have to specify again all inputs parameters?
is it not possible to take the parameters from the last run?

**Reply by terrytangyuan** [reply]
It's only used if you want to override your previously provided parameter values
